contrib/src/main/java/gov/nasa/jpl/aerie/contrib/models/SampledResource.java

```diff
@@ -1,35 +1,68 @@
 package gov.nasa.jpl.aerie.contrib.models;
 
 import gov.nasa.jpl.aerie.merlin.framework.resources.discrete.DiscreteResource;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
 
 import java.util.Objects;
 import java.util.function.Supplier;
 import java.util.function.UnaryOperator;
 
 import static gov.nasa.jpl.aerie.merlin.framework.ModelActions.*;
-import static gov.nasa.jpl.aerie.merlin.protocol.types.Duration.SECOND;
 
+/**
+ * Simple resource that samples arbitrarily many existing resources/values at a specified rate (default rate is once per
+ * second).
+ */
 public class SampledResource<T> implements DiscreteResource<T> {
   private final Register<T> result;
   private final Supplier<T> sampler;
+  private double rate;
```

Comment:
Strictly speaking, we're talking about a "period" rather than a "rate" - any thoughts on what word would cause the least confusion?

Reply:
Good point, we could also do something as simple as "dt"

Reply:
I went your suggested term, period.